homeassistant/components/livisi/climate.py

```diff
@@ -0,0 +1,214 @@
+    coordinator: LivisiDataUpdateCoordinator = hass.data[DOMAIN][config_entry.entry_id]
+
+    @callback
+    def handle_coordinator_update() -> None:
+        """Add climate device."""
+        shc_devices: list[dict[str, Any]] = coordinator.data
+        entities: list[ClimateEntity] = []
+        for device in shc_devices:
+            if (
+                device["type"] == VRCC_DEVICE_TYPE
+                and device["id"] not in coordinator.devices
+            ):
+                livisi_climate: ClimateEntity = create_entity(
+                    config_entry, device, coordinator
+                )
+                LOGGER.debug("Include device type: %s", device.get("type"))
+                coordinator.devices.add(device["id"])
+                entities.append(livisi_climate)
+        async_add_entities(entities)
+
+    config_entry.async_on_unload(
+        coordinator.async_add_listener(handle_coordinator_update)
+    )
+
+
+def create_entity(
+    config_entry: ConfigEntry,
+    device: dict[str, Any],
+    coordinator: LivisiDataUpdateCoordinator,
+) -> ClimateEntity:
+    """Create Climate Entity."""
+    capabilities: Mapping[str, Any] = device[CAPABILITY_MAP]
+    room_id: str = device["location"]
+    room_name: str = coordinator.rooms[room_id]
+    livisi_climate = LivisiClimate(
+        config_entry,
+        coordinator,
+        unique_id=device["id"],
+        manufacturer=device["manufacturer"],
+        device_type=device["type"],
+        target_temperature_capability=capabilities["RoomSetpoint"],
+        temperature_capability=capabilities["RoomTemperature"],
+        humidity_capability=capabilities["RoomHumidity"],
+        room=room_name,
+    )
+    return livisi_climate
+
+
+class LivisiClimate(CoordinatorEntity[LivisiDataUpdateCoordinator], ClimateEntity):
+    """Represents the Livisi Climate."""
+
+    def __init__(
+        self,
+        config_entry: ConfigEntry,
+        coordinator: LivisiDataUpdateCoordinator,
+        unique_id: str,
+        manufacturer: str,
+        device_type: str,
+        target_temperature_capability: str,
+        temperature_capability: str,
+        humidity_capability: str,
+        room: str,
+    ) -> None:
+        """Initialize the Livisi Climate."""
+        self.config_entry = config_entry
+        self._attr_unique_id = unique_id
+        self._target_temperature_capability = target_temperature_capability
+        self._temperature_capability = temperature_capability
+        self._humidity_capability = humidity_capability
+        self.aio_livisi = coordinator.aiolivisi
+        self._attr_available = False
+        self._attr_hvac_modes = [HVACMode.HEAT]
+        self._attr_hvac_mode = HVACMode.HEAT
+        self._attr_temperature_unit = UnitOfTemperature.CELSIUS
+        self._attr_supported_features = ClimateEntityFeature.TARGET_TEMPERATURE
+        self._attr_target_temperature_high = MAX_TEMPERATURE
+        self._attr_target_temperature_low = MIN_TEMPERATURE
```

Comment:
Since these 6 attributes are fixed, then should be set as class attributes not instance attributes:
```python
class LivisiClimate(CoordinatorEntity[LivisiDataUpdateCoordinator], ClimateEntity):
    """Represents the Livisi Climate."""

    _attr_hvac_modes = [HVACMode.HEAT]
    _attr_hvac_mode = HVACMode.HEAT
    _attr_temperature_unit = UnitOfTemperature.CELSIUS
    _attr_supported_features = ClimateEntityFeature.TARGET_TEMPERATURE
    _attr_target_temperature_high = MAX_TEMPERATURE
    _attr_target_temperature_low = MIN_TEMPERATURE
```

Reply:
I added this change, thank you.

Reply:
The six lines need to be moved above `__init__` method.

Reply:
I'm sorry for the confusion, I changed this.